src/lib.rs

```diff
@@ -18,10 +36,29 @@ extern crate libc;
 
 
 /// Returns the number of available CPUs of the current system.
+/// 
+/// This function will get the number of logical cores. Sometimes this is different to the number
+/// of physical cores becuase the cpu chip can deliver better performance when using more threads.
```

Comment:
Whether SMT actually delivers more performance or not is [debatable](https://en.wikipedia.org/wiki/Simultaneous_multithreading#Disadvantages).

I think we should just stick to saying that some systems have multiple logical cores per physical CPU core, and link to "Simultaneous multithreading" as an example.

Reply:
Agreed.